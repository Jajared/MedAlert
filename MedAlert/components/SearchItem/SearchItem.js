import React, { PureComponent } from "react";
import { StyleSheet, View, Text, TouchableOpacity, Image } from "react-native";
import { AntDesign, MaterialCommunityIcons, Feather } from "@expo/vector-icons";

const dosageFormIcons = {
  Tablet: <Image source={require("../../assets/medicationDB/tablet.png")} style={{ width: 24, height: 24 }} />,
  Injection: <Image source={require("../../assets/medicationDB/injection.png")} style={{ width: 24, height: 24 }} />,
  Capsule: <Image source={require("../../assets/medicationDB/capsule.png")} style={{ width: 24, height: 24 }} />,
  Cream: <Image source={require("../../assets/medicationDB/cream.png")} style={{ width: 24, height: 24 }} />,
  Solution: <Image source={require("../../assets/medicationDB/solution.png")} style={{ width: 24, height: 24 }} />,
  Granule: <Image source={require("../../assets/medicationDB/capsule.png")} style={{ width: 24, height: 24 }} />,
  Syrup: <Image source={require("../../assets/medicationDB/syrup.png")} style={{ width: 24, height: 24 }} />,
  Ointment: <Image source={require("../../assets/medicationDB/ointment.png")} style={{ width: 24, height: 24 }} />,
  Powder: <Image source={require("../../assets/medicationDB/powder.png")} style={{ width: 24, height: 24 }} />,
};

export default class SearchItem extends PureComponent {
  getDosageFormIcon(dosageForm) {
    const formattedDosageForm = dosageForm.toLowerCase().replace(/\s/g, "");

    for (const form in dosageFormIcons) {
      if (formattedDosageForm.includes(form.toLowerCase())) {
        return dosageFormIcons[form];
      }
    }

    // If no matching dosage form found, return a default icon
    return <AntDesign name="medicinebox" size={24} color="black" />;
  }

  render() {
    const { item, navigation } = this.props;
    return (
      <TouchableOpacity onPress={() => navigation.navigate("Search Medication Item", { medicationDetails: item })} style={styles.searchItem}>
        <View style={styles.searchItemIcon}>{this.getDosageFormIcon(item.dosage_form)}</View>
        <View style={styles.searchItemText}>
          <Text style={styles.searchItemTitle}>{item.product_name}</Text>
          <Text style={styles.seachItemInfo}>{item.manufacturer}</Text>
        </View>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  searchItem: {
    flex: 1,
    flexDirection: "row",
    padding: 10,
    marginVertical: 8,
    marginHorizontal: 5,
    borderColor: "black",
    borderWidth: 1,
    borderRadius: 8,
    backgroundColor: "#f0f0f0",
  },
  searchItemIcon: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  searchItemText: {
    flex: 7,
  },
  searchItemInfo: {
    fontSize: 12,
  },
  searchItemTitle: {
    fontSize: 15,
    fontWeight: "bold",
  },
});
