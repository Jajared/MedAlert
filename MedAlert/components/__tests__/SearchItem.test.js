import { render, screen, fireEvent } from "@testing-library/react-native";
import SearchItem from "../SearchItem";
import { expect, jest, test } from "@jest/globals";

jest.mock("expo-font");
jest.mock("expo-asset");

const navigationObject = {
  navigate: jest.fn(),
  goBack: jest.fn(),
};

const testItem = {
  product_name: "Test Product",
  manufacturer: "Test Manufacturer",
  dosage_form: "Test Dosage Form",
};

describe("SearchItem", () => {
  // Check if renders correctly
  it("renders correctly", () => {
    const tree = render(<SearchItem navigation={navigationObject} item={testItem} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  // Check number of children
  it("has 3 children", () => {
    render(<SearchItem navigation={navigationObject} item={testItem} />);
    expect(screen.getByTestId("SearchItem").children.length).toBe(3);
  });
  // Check if navigates to search item page when pressed
  it("Navigates to search item page", () => {
    render(<SearchItem navigation={navigationObject} item={testItem} />);
    fireEvent.press(screen.getByTestId("SearchItem"));
    expect(navigationObject.navigate).toBeCalledWith("Search Medication Item", { medicationDetails: testItem });
  });
});
