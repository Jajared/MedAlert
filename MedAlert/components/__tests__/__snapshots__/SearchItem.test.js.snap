// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchItem renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "white",
      "borderBottomColor": "#E0E0E0",
      "borderBottomWidth": 2,
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="SearchItem"
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text />
  </View>
  <View
    style={
      {
        "flex": 7,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 15,
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }
    >
      Test Product
    </Text>
    <Text
      style={
        {
          "fontSize": 12,
        }
      }
    >
      Test Manufacturer
    </Text>
  </View>
</View>
`;
